x-pack/legacy/plugins/reporting/server/browsers/chromium/driver_factory/index.ts

```diff
@@ -13,80 +13,79 @@ import {
   ConsoleMessage,
   Request as PuppeteerRequest,
 } from 'puppeteer';
+import { ChildProcess } from 'child_process';
 import rimraf from 'rimraf';
 import * as Rx from 'rxjs';
 import { ignoreElements, mergeMap, tap } from 'rxjs/operators';
 import { InnerSubscriber } from 'rxjs/internal/InnerSubscriber';
 
-import { puppeteerLaunch } from '../puppeteer';
+import { BrowserConfig } from '../../../../types';
 import { LevelLogger as Logger } from '../../../lib/level_logger';
 import { HeadlessChromiumDriver } from '../driver';
-import { args, IArgOptions } from './args';
 import { safeChildProcess } from '../../safe_child_process';
+import { puppeteerLaunch } from '../puppeteer';
 import { getChromeLogLocation } from '../paths';
+import { args } from './args';
 
 type binaryPath = string;
 type queueTimeout = number;
-interface IBrowserConfig {
-  [key: string]: any;
-}
 
 export class HeadlessChromiumDriverFactory {
   private binaryPath: binaryPath;
   private logger: Logger;
-  private browserConfig: IBrowserConfig;
+  private browserConfig: BrowserConfig;
   private queueTimeout: queueTimeout;
 
   constructor(
     binaryPath: binaryPath,
     logger: Logger,
-    browserConfig: IBrowserConfig,
+    browserConfig: BrowserConfig,
     queueTimeout: queueTimeout
   ) {
     this.binaryPath = binaryPath;
     this.browserConfig = browserConfig;
     this.queueTimeout = queueTimeout;
-    this.logger = logger;
+    this.logger = logger; // TODO: just pass logger into each method from outside, like test() has it
   }
 
   type = 'chromium';
 
-  test(
-    { viewport, browserTimezone }: { viewport: IArgOptions['viewport']; browserTimezone: string },
-    logger: Logger
-  ) {
+  test({ viewport }: { viewport: BrowserConfig['viewport'] }, logger: Logger) {
     const userDataDir = fs.mkdtempSync(path.join(os.tmpdir(), 'chromium-'));
     const chromiumArgs = args({
       userDataDir,
       viewport,
       verboseLogging: true,
       disableSandbox: this.browserConfig.disableSandbox,
-      proxyConfig: this.browserConfig.proxy,
+      proxy: this.browserConfig.proxy,
     });
 
     return puppeteerLaunch({
       userDataDir,
       executablePath: this.binaryPath,
       ignoreHTTPSErrors: true,
       args: chromiumArgs,
-      env: {
-        TZ: browserTimezone,
-      },
-    } as LaunchOptions).catch((error: Error) => {
-      logger.error(
-        `The Reporting plugin encountered issues launching Chromium in a self-test. You may have trouble generating reports.`
-      );
-      logger.error(error);
-      logger.warning(`See Chromium's log output at "${getChromeLogLocation(this.binaryPath)}"`);
-      return null;
-    });
+    } as LaunchOptions)
+      .then((browser: Browser) => {
+        const childProcess: ChildProcess = browser.process();
+        logger.debug(`Test browser process launched with PID: ${childProcess.pid}`);
```

Comment:
This is 1/2 of the actual changes. Everything else is due to increase strictness of Typescript